### PR TITLE
send an email when the solicited review is declined

### PR DESCRIPTION
**Pull request by melisabok**

### CODE REVIEW

File: openreview/journal/journal.py [hunk 1]
```diff
@@ -449,7 +452,12 @@ def get_late_invitees(self, invitation_id):
                     signature_members.append(signature)
                 else:
                     signature_members = signature_members + signature['members']
+            for signature in reply.signatures:
```
Comment (melisabok) [inline openreview/journal/journal.py:455]: @carlosmondra this is the fix I'm doing.

Comment (melisabok) [reply openreview/journal/journal.py:455]: This code is to check who are the late invitees. in the case where the signature is a tilde id, I don't need to check the signature in details, I can check the reply.signatures instead.

File: tests/test_journal.py [hunk 2]
```diff
@@ -2648,6 +2648,43 @@ def test_submitted_submission(self, journal, openreview_client, helpers):
             weight=1
         ))
 
+        ## Ask solitic review with a conflict
+        tom_client = OpenReviewClient(username='tom@mail.com', password='1234')
+        solitic_review_note = tom_client.post_note_edit(invitation=f'{venue_id}/Paper7/-/Solicit_Review',
```
Comment (celestemartinez) [inline tests/test_journal.py:2653]:
```suggestion
        solicit_review_note = tom_client.post_note_edit(invitation=f'{venue_id}/Paper7/-/Solicit_Review',
```

File: tests/test_journal.py [hunk 3]
```diff
@@ -2648,6 +2648,43 @@ def test_submitted_submission(self, journal, openreview_client, helpers):
...
+            signatures=['~Tom_Rain1'],
+            note=Note(
+                content={
+                    'solicit': { 'value': 'I solicit to review this paper.' },
+                    'comment': { 'value': 'I can review this paper.' }
+                }
+            )
+        )
+
+        helpers.await_queue_edit(openreview_client, edit_id=solitic_review_note['id'])
```
Comment (celestemartinez) [inline tests/test_journal.py:2663]:
```suggestion
        helpers.await_queue_edit(openreview_client, edit_id=solicit_review_note['id'])
```

File: tests/test_journal.py [hunk 4]
```diff
@@ -2648,6 +2648,43 @@ def test_submitted_submission(self, journal, openreview_client, helpers):
...
+
+        ## Post a response
+        solitic_review_approval_note = joelle_client.post_note_edit(invitation=f'{venue_id}/Paper7/-/~Tom_Rain1_Solicit_Review_Approval',
```
Comment (celestemartinez) [inline tests/test_journal.py:2666]:
```suggestion
        solicit_review_approval_note = joelle_client.post_note_edit(invitation=f'{venue_id}/Paper7/-/~Tom_Rain1_Solicit_Review_Approval',
```

File: tests/test_journal.py [hunk 5]
```diff
@@ -2648,6 +2648,43 @@ def test_submitted_submission(self, journal, openreview_client, helpers):
...
+            signatures=[f"{venue_id}/Paper7/Action_Editors"],
+            note=Note(
+                forum=note_id_7,
+                replyto=solitic_review_note['note']['id'],
+                content={
+                    'decision': { 'value': 'No, I decline the solitic review.' },
```
Comment (celestemartinez) [inline tests/test_journal.py:2672]:
```suggestion
                    'decision': { 'value': 'No, I decline the solicit review.' },
```

File: tests/test_journal.py [hunk 6]
```diff
@@ -2648,6 +2648,43 @@ def test_submitted_submission(self, journal, openreview_client, helpers):
...
+                    'comment': { 'value': 'Sorry, all the reviewers were assigned.' }
+                }
+            )
+        )
+
+        helpers.await_queue_edit(openreview_client, edit_id=solitic_review_approval_note['id'])
```
Comment (celestemartinez) [inline tests/test_journal.py:2678]:
```suggestion
        helpers.await_queue_edit(openreview_client, edit_id=solicit_review_approval_note['id'])
```

File: tests/test_journal.py [hunk 7]
```diff
@@ -2648,6 +2648,43 @@ def test_submitted_submission(self, journal, openreview_client, helpers):
...
+
+        messages = journal.client.get_messages(to = 'tom@mail.com', subject = '[TMLR] Request to review TMLR submission "Paper title 7" was not accepted')
+        assert len(messages) == 1
+        assert messages[0]['content']['text'] == f'''<p>Hi Tom Rain,</p>
+<p>This is to inform you that your request to act as a reviewer for TMLR submission Paper title 7 was not accepted by the Action Editor (AE). If you would like to know more about the reason behind this decision, you can click here: <a href=\"https://openreview.net/forum?id={note_id_7}&amp;noteId={solitic_review_approval_note['note']['id']}\">https://openreview.net/forum?id={note_id_7}&amp;noteId={solitic_review_approval_note['note']['id']}</a>.</p>
```
Comment (celestemartinez) [inline tests/test_journal.py:2683]:
```suggestion
<p>This is to inform you that your request to act as a reviewer for TMLR submission Paper title 7 was not accepted by the Action Editor (AE). If you would like to know more about the reason behind this decision, you can click here: <a href=\"https://openreview.net/forum?id={note_id_7}&amp;noteId={solicit_review_approval_note['note']['id']}\">https://openreview.net/forum?id={note_id_7}&amp;noteId={solicit_review_approval_note['note']['id']}</a>.</p>
```

File: openreview/journal/process/solicit_review_approval_process.py [hunk 8]
```diff
@@ -36,6 +36,27 @@ def process(client, edit, invitation):
 
 The {journal.short_name} Editors-in-Chief
 
+''',
+            replyTo=journal.contact_info
+        )
+
+        return
+
+    if note.content['decision']['value'] == 'No, I decline the solitic review.':
```
Comment (celestemartinez) [inline openreview/journal/process/solicit_review_approval_process.py:45]:
```suggestion
    if note.content['decision']['value'] == 'No, I decline the solicit review.':
```

**Review by celestemartinez — APPROVED** [review]

File: tests/test_journal.py [hunk 9]
```diff
@@ -2648,6 +2648,43 @@ def test_submitted_submission(self, journal, openreview_client, helpers):
...
+        )
+
+        helpers.await_queue_edit(openreview_client, edit_id=solicit_review_note['id'])
+
+        ## Post a response
+        solicit_review_approval_note = joelle_client.post_note_edit(invitation=f'{venue_id}/Paper7/-/~Tom_Rain1_Solicit_Review_Approval',
+            signatures=[f"{venue_id}/Paper7/Action_Editors"],
+            note=Note(
+                forum=note_id_7,
+                replyto=solitic_review_note['note']['id'],
```
Comment (celestemartinez) [inline tests/test_journal.py:2670]:
```suggestion
                replyto=solicit_review_note['note']['id'],
```
I missed one!

**Pull request merged**